### PR TITLE
Add new samples

### PR DESCRIPTION
**Pull request by bamx23**
This PR adds a minimal sample for each of the supported platforms.

Supporting Xcode projects for each of the platform is a bit painful, so here I'm adding a dependency on Tuist for projects generation. 
The common part of the samples is managed using SPM to ensure that integration of KSCrash into samples is as close as possible to real use of SPM. Sorry, no CocoaPods (not sorry).

As a follow up, we can create a set of build jobs to ensure there's no breaking API changes (after most of the APIs are used in these samples).

### CODE REVIEW

File: Samples/Common/Package.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// swift-tools-version:5.3
+
+import PackageDescription
+
+let package = Package(
+  name: "KSCrashSamplesCommon",
+  platforms: [
```
Comment (GLinnik21) [inline Samples/Common/Package.swift:7]: xrOS?

File: Samples/Common/Package.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ),
+  ],
+  dependencies: [
+    .package(path: "../.."),
+  ],
+  targets: [
+    .target(
+      name: "LibraryBridge",
+      dependencies: [
+        .productItem(name: "Recording", package: "KSCrash"),
```
Comment (GLinnik21) [inline Samples/Common/Package.swift:34]: Is `productItem` a modern API?

File: Samples/Common/Sources/LibraryBridge/RecordingSample.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+import Foundation
+import KSCrashRecording
+
+public class RecordingSample {
+    public static func simpleInstall() {
+        let config = KSCrashConfiguration()
```
Comment (GLinnik21) [inline Samples/Common/Sources/LibraryBridge/RecordingSample.swift:32]: What do you think? Does this name look ok in swift?

File: Samples/Common/Sources/SampleUI/SampleView.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import SwiftUI
+import LibraryBridge
+import CrashTriggers
+
+public struct SampleView: View {
+    public init() { }
+    
+    public var body: some View {
+        VStack {
```
Comment (GLinnik21) [inline Samples/Common/Sources/SampleUI/SampleView.swift:35]: Maybe better to use `List` here?

**Review by GLinnik21 — COMMENTED** [review]
Actually I thought that we can use one SwiftUI code for all the platforms without separate subfolders and configurations. If it's not applicable to Tuist, maybe one xcproject is not a bad solution?

**Comment by bamx23** [comment]
> Actually I thought that we can use one SwiftUI code for all the platforms without separate subfolders and configurations. If it's not applicable to Tuist, maybe one xcproject is not a bad solution?

While I think it can be possible, I would still have this entry point file to be separate for each of the samples. The most of the code is intended to go to that common SwiftUI library. But it's already not the same for macOS as we need to override the app delegate. 2 files per sample that you don't need to touch in most of the cases – sounds good to me.

File: Samples/Common/Sources/LibraryBridge/RecordingSample.swift (hunk 3 above)
Comment (bamx23) [reply Samples/Common/Sources/LibraryBridge/RecordingSample.swift:32]: Not sure. I would want to have just `install` if I don't care about config. 

And also maybe having a way to inline config creation like:
```swift
KSCrash.shared().install(with: .init(monitors: [.asyncSafe]))
```
But this last idea is the one I'm not sure about here.

File: Samples/Common/Package.swift (hunk 2 above)
Comment (bamx23) [reply Samples/Common/Package.swift]: Good catch, autocomplete confused me

**Review by GLinnik21 — APPROVED** [review]

**Pull request merged**